cumulus/client/service/src/lib.rs

```diff
@@ -580,3 +607,144 @@ where
+						},
+					};
+					let backed_block_time = Instant::now();
+					if let Some(last_backed_block_time) = &last_backed_block_time {
+						let duration = backed_block_time.duration_since(*last_backed_block_time);
+						if let Some(metrics) = &metrics {
+							metrics
+								.parachain_block_authorship_duration
+								.observe(duration.as_secs_f64());
+						}
+					}
+					last_backed_block_time = Some(backed_block_time);
+					backed_candidates.push(backed_block);
+				},
+				CandidateEvent::CandidateIncluded(_, head, _, _) => {
+					let included_block = match Block::Header::decode(&mut &head.0[..]) {
+						Ok(header) => header,
+						Err(e) => {
+							log::warn!(
+								"Failed to decode parachain header from included block: {e:?}"
+							);
+							continue
+						},
+					};
+					let unincluded_segment_size =
+						client.info().best_number.saturating_sub(*included_block.number());
+					let unincluded_segment_size: u32 = unincluded_segment_size.saturated_into();
+					if let Some(metrics) = &metrics {
+						metrics.unincluded_segment_size.observe(unincluded_segment_size.into());
+					}
+					included_candidates.push(included_block);
+				},
+				CandidateEvent::CandidateTimedOut(_, head, _) => {
+					let timed_out_block = match Block::Header::decode(&mut &head.0[..]) {
+						Ok(header) => header,
+						Err(e) => {
+							log::warn!(
+								"Failed to decode parachain header from timed out block: {e:?}"
+							);
+							continue
+						},
+					};
+					timed_out_candidates.push(timed_out_block);
+				},
+			}
+		}
+		let mut log_parts = Vec::new();
+		if !backed_candidates.is_empty() {
+			let backed_candidates = backed_candidates
+				.into_iter()
+				.map(|c| format!("#{} ({})", c.number(), c.hash()))
+				.collect::<Vec<_>>()
+				.join(", ");
+			log_parts.push(format!("backed: {}", backed_candidates));
+		};
+		if !included_candidates.is_empty() {
+			let included_candidates = included_candidates
+				.into_iter()
+				.map(|c| format!("#{} ({})", c.number(), c.hash()))
+				.collect::<Vec<_>>()
+				.join(", ");
+			log_parts.push(format!("included: {}", included_candidates));
+		};
+		if !timed_out_candidates.is_empty() {
+			let timed_out_candidates = timed_out_candidates
+				.into_iter()
+				.map(|c| format!("#{} ({})", c.number(), c.hash()))
+				.collect::<Vec<_>>()
+				.join(", ");
+			log_parts.push(format!("timed out: {}", timed_out_candidates));
+		};
+		if !log_parts.is_empty() {
+			log::info!(
+				"Update at relay chain block #{} ({}) - {}",
+				n.number(),
+				n.hash(),
+				log_parts.join(", ")
+			);
+		}
+	}
+}
+
+struct ParachainInformantMetrics {
+	parachain_block_authorship_duration: Histogram,
+	unincluded_segment_size: Histogram,
+}
+
+impl ParachainInformantMetrics {
+	fn new(prometheus_registry: &Registry) -> prometheus::Result<Self> {
+		let parachain_block_authorship_duration = Histogram::with_opts(HistogramOpts::new(
+			"parachain_block_authorship_duration",
```

Comment:
This sounds wrong, should be `parachain_block_backed_duration` or something. Authorship implies this is only about authoring.